RealmSwift/RealmConfiguration.swift

```diff
@@ -200,7 +200,11 @@ extension Realm {
         /// The classes managed by the Realm.
         public var objectTypes: [ObjectBase.Type]? {
             get {
-                return self.customSchema.map { $0.objectSchema.compactMap { $0.objectClass as? Object.Type } }
+                return self.customSchema.map {
+                    $0.objectSchema.compactMap {
+                        $0.objectClass as? Object.Type ?? $0.objectClass as? EmbeddedObject.Type
```

Comment:
Can we just leave a comment here saying what this is doing? It's fairly obvious when reading it with context, but it certainly looks strange.